File: library/src/main/java/com/ashlikun/loadswitch/OnLoadLayoutListener.java
package com.ashlikun.loadswitch;

import android.view.View;

/**
 * 作者　　: 李坤
 * 创建时间: 17:08 Administrator
 * 邮箱　　：496546144@qq.com
 * <p>
 * 功能介绍：布局切换时的监听
 */
public interface OnLoadLayoutListener {

    void setRetryEvent(View retryView, ContextData data);

    void setLoadingEvent(View loadingView, ContextData data);

    void setEmptyEvent(View emptyView, ContextData data);
}

File: library/src/main/java/com/ashlikun/loadswitch/OnLoadLayoutListener.kt
package com.ashlikun.loadswitch

import android.view.View

/**
 * @author　　: 李坤
 * 创建时间: 2022/4/30 17:04
 * 邮箱　　：496546144@qq.com
 *
 * 功能介绍：布局切换时的监听
 */
interface OnLoadLayoutListener {
    fun setRetryEvent(retryView: View, data: ContextData)
    fun setLoadingEvent(loadingView: View, data: ContextData)
    fun setEmptyEvent(emptyView: View, data: ContextData)
}